### PR TITLE
Activity concept rewrite

### PR DESCRIPTION
**Pull request by cullywakelin**
This PR rewrites the Activity concept information.

Information added:
- overview diagram
- activity option information

Information removed:
Anything that can/should be converted to a "How to" guide such as
- cancellation
- task routing
- leader election

### CODE REVIEW

File: docs/activities.md [hunk 1]
```diff
@@ -1,81 +1,69 @@
...
+### Wait for cancellation
 
-Another common case for activity heartbeat failure is that the workflow that invoked it is in a completed state. In this case an activity is expected to perform cleanup as well.
+When an Activity is cancelled it may be necessary for the Activity to revert actions that it has already taken. This option specifies to the Workflow whether to wait for the clean up to occur after an Activity has been cancelled.
 
-## Activity Task Routing through Task Queues
+### Activity Id
 
-Activities are dispatched to workers through task queues. Task queues are queues that workers listen on. Task queues are highly dynamic and lightweight. They don't need to be explicitly registered. And it is okay to have one task queue per worker process. It is normal to have more than one activity type to be invoked through a single task queue. And it is normal in some cases (like host routing) to invoke the same activity type on multiple task queues.
+Use this to give your Activity a name or unique identifier that correlates to your business logic. This is optional and is not used by the Temporal server.
```
Comment (rylandg) [inline docs/activities.md:45]: 1. "this"
2. I would use the word mapping and "relevant business ID" or something

File: docs/activities.md [hunk 2]
```diff
@@ -1,81 +1,69 @@
...
-- _Workers with different capabilities_. For example, workers on GPU boxes vs non GPU boxes. Having two separate task queues in this case allows workflows to pick which one to send activity an execution request to.
-- _Routing activity to a specific host_. For example, in the media encoding case the transform and upload activity have to run on the same host as the download one.
-- _Routing activity to a specific process_. For example, some activities load large data sets and caches it in the process. The activities that rely on this data set should be routed to the same process.
-- _Multiple priorities_. One task queue per priority and having a worker pool per priority.
-- _Versioning_. A new backwards incompatible implementation of an activity might use a different task queue.
+Unlike Workflows, Temporal does not recover the state of an Activity. Activity failures are expected to occur and when they do the Activity can be automatically retried per its Retry policy. For practical implementation guidance, read [How to retry Activities and Workflows](retry-activities-and-workflows).
```
Comment (rylandg) [inline docs/activities.md:49]: "Activity failures are expected to occur " - 100% true but phrase it more positively please. Not trying to hide the truth but it sort of seems like it's due to Temporal with current structure

File: docs/activities.md [hunk 3]
```diff
@@ -1,81 +1,69 @@
...
 
-## Asynchronous Activity Completion
+## Local Activities
 
-By default an activity is a function or a method depending on a client side library language. As soon as the function returns, an activity completes. But in some cases an activity implementation is asynchronous. For example it is forwarded to an external system through a message queue. And the reply comes through a different queue.
+For Activities that have perfunctory requirements, Temporal supports a stripped down version of Activity implementations called "local Activities". A local Activity is executed by the same Worker that is executing the Workflow that invoked it. A function can be executed as a local Activity if it:
```
Comment (rylandg) [inline docs/activities.md:53]: 1. What is a "perfunctory"
2. Why is it stripped down? What is stripped down anywa?
3. Rephrase "A local Activity is executed by the same Worker that is executing the Workflow that invoked it"

File: docs/activities.md [hunk 4]
```diff
@@ -1,81 +1,69 @@
...
 
-To support such use cases, Temporal allows activity implementations that do not complete upon activity function completions. A separate API should be used in this case to complete the activity. This API can be called from any process, even in a different programming language, that the original activity worker used.
+- takes no more than a few seconds to complete.
```
Comment (rylandg) [inline docs/activities.md:55]: avoid negation phrasing - only takes a few seconds to complete

File: docs/activities.md [hunk 5]
```diff
@@ -1,81 +1,69 @@
...
+- does not require global rate limiting.
+- does not require routing to specific Workers or pools of Workers.
+- can be implemented in the same binary as the Workflow that invokes it.
```
Comment (rylandg) [inline docs/activities.md:58]: not sure how to make this more digestible, think about ti

File: docs/activities.md [hunk 6]
```diff
@@ -1,81 +1,69 @@
...
 
-## Local Activities
+The main benefit of local Activities is that they are much more efficient in utilizing Temporal service resources and have much lower latency overhead in comparison to the usual Activity invocation.
```
Comment (rylandg) [inline docs/activities.md:60]: " to the usual Activity invocation." - rephrase

Comment (rylandg) [reply docs/activities.md:60]: Also "much lower latency" is not optimal, what is "much"

File: docs/activities.md [hunk 7]
```diff
@@ -1,81 +1,69 @@
...
+
+### Local Activity options
 
-Some of the activities are very short lived and do not need the queing semantic, flow control, rate limiting and routing capabilities. For these Temporal supports so called _local activity_ feature. Local activities are executed in the same worker process as the workflow that invoked them. Consider using local activities for functions that are:
+Local activities support a small subset of options:
```
Comment (rylandg) [inline docs/activities.md:64]:
```suggestion
Local activities support a small subset of Activity options:
```

**Comment by cullywakelin** [comment]
@rapmastersethmc will be redesigning and rewriting this content area

**Pull request closed without merging**